src/coreclr/vm/eepolicy.cpp

```diff
@@ -371,34 +373,41 @@ void LogInfoForFatalError(UINT exitCode, LPCWSTR pszMessage, PEXCEPTION_POINTERS
 
     EX_TRY
     {
+        StackSString message;
         if (exitCode == (UINT)COR_E_FAILFAST)
         {
-            PrintToStdErrA("Process terminated. ");
+            message.Append(W("Process terminated. "));
```

Comment:
One last thing. I would still prefer these two short messages to be printed separately to get at least these to the output when things are in a really bad state. 
The rest looks good.

Reply:
OK, for Android that will mean they will end up on a separate log line compared to how things are on stdout, but maybe it would make sense to actually do them on separate line always, so do "Process terminated.\n".

Reply:
For this message, I think it doesn't really matter, it maybe even easier to read in the log. The `errorSource` is usually not set. I think the only caller that ever sets it is the managed `Environment.FailFast` and only when called from within System.Private.CoreLib.dll. And frankly, I am not sure if there are any calls that would pass it in.

Reply:
I ended up adding support to our Android logging to ignore a logging that is just "\n", meaning that its OK, to log something, like a preallocated string and then fire a "\n" like we did in this example, that will end up as one log line both in logcat and stdout. With that we can keep old behavior of this logging, just hitting stack usage when need to convert from unicode -> mb and when converting error code to message through GetHRMsg.